osgp/platform/osgp-secret-management/src/main/java/org/opensmartgridplatform/secretmanagement/application/services/SecretManagementService.java

```diff
@@ -0,0 +1,164 @@
+        final Date now = new Date();
+        final Page<DbEncryptionKeyReference> keyRefsPage = this.keyRepository.findByTypeAndValid(now,
+                this.encryptionProviderType, Pageable.unpaged());
+        if (keyRefsPage.getSize() > 1) {
+            throw new IllegalStateException("Multiple encryption keys found that are valid at " + now);
+        }
+        return keyRefsPage.stream().findFirst().orElseThrow(
+                () -> new NoSuchElementException("No encryption key found that is valid at " + now));
+    }
+
+    private TypedSecret validateSecret(final String deviceIdentification, final TypedSecret secret) {
+        if (secret.getSecret() == null) {
+            throw new IllegalArgumentException("No secret string set");
+        } else if (secret.getSecretType() == null) {
+            throw new IllegalArgumentException("No secret type set");
+        } else if (this.isIdenticalToCurrent(deviceIdentification, secret)) {
+            throw new IllegalArgumentException(
+                    String.format("Secret is identical to current secret (%s, %s)", deviceIdentification,
+                            secret.getSecretType().name()));
+        }
+        return secret;
+    }
+
+    private boolean isIdenticalToCurrent(final String deviceIdentification, final TypedSecret secret) {
+        final Optional<TypedSecret> current = this.retrieveSecret(deviceIdentification, secret.getSecretType());
+        return current.isPresent() ? current.get().getSecret().equals(secret.getSecret()) : false;
+    }
+
+    private DbEncryptedSecret createEncrypted(final String deviceIdentification, final TypedSecret typedSecret,
+            final DbEncryptionKeyReference keyReference) {
+        final String secretString = typedSecret.getSecret();
+        final byte[] secretBytes = HexUtils.fromHexString(secretString);
+        final Secret secret = new Secret(secretBytes);
+        try {
+            final EncryptedSecret encryptedSecret = this.encryptionDelegate.encrypt(
+                    keyReference.getEncryptionProviderType(), secret, keyReference.getReference());
+            final DbEncryptedSecret dbEncryptedSecret = new DbEncryptedSecret();
+            dbEncryptedSecret.setDeviceIdentification(deviceIdentification);
+            dbEncryptedSecret.setEncodedSecret(HexUtils.toHexString(encryptedSecret.getSecret()));
+            dbEncryptedSecret.setSecretType(typedSecret.getSecretType());
+            dbEncryptedSecret.setEncryptionKeyReference(keyReference);
+            dbEncryptedSecret.setCreationTime(new Date());
+            return dbEncryptedSecret;
+        } catch (final Exception exc) {
+            throw new IllegalStateException("Could not create encrypted secret", exc);
+        }
+    }
+
+    public List<TypedSecret> retrieveSecrets(final String deviceIdentification, final List<SecretType> secretTypes)
+            throws Exception {
+        try {
+            //@formatter:off
```

Comment:
Your IDE should be able to format this nicely automatically (the same goes for the previous "@formatter" code.